### PR TITLE
Logout

### PR DESCRIPTION
**Pull request by carperimb**
- 2 Tests cases were implemented.
- Documentation updated.
- Login TCs were skipped just for this sprint.

### CODE REVIEW

File: pages/logout.page.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+
+import BasePage from '../pages/base.page';
+
+class LogoutPage extends BasePage {
+    //WebElements
+    get userAvatar(){return $('[class="user-avatar rounded-circle"]')};
```
Comment (rquintana-abstracta) [inline pages/logout.page.js]: Prefer using only `user-avatar` class, if that's enough for locating that element, as `rounded-circle` seems to be a more general class, used exclusively for styling purposes, and can potentially change faster than user-avatar (if they decide to style it differently, for example), making your selector more flaky / less robust.

File: tests/login.test.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { expect } from 'chai';
 describe('OpenBank', () => {
 
     
-    it('Should access the home page when attempting to login with valid credentials', async () => {
+    it.skip('Should access the home page when attempting to login with valid credentials', async () => {
```
Comment (rquintana-abstracta) [inline tests/login.test.js]: Is there a reason for skipping all login tests?

File: tests/logout.test.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import logoutPage from '../pages/logout.page';
+
+describe('OpenBank', () => {
+
+    it('Should show logout success message when clicking on logout menu', async () => {
+
+      await loginPage.open('login');
+      let username = 'jsmith@demo.io';
+      let password = 'Demo123!';
+      await loginPage.logIn(username, password);
```
Comment (rquintana-abstracta) [inline tests/logout.test.js]: This logic seems to be a pre-requisite for all logout tests, and it's currently duplicated. Consider using a `beforeEach()` **hook** to put it there, so you define it once, and it'll run automatically before each test in the suite, removing this duplication.

File: tests/logout.test.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+      await logoutPage.clickAvatar();
+      
+      expect(await browser.getUrl()).to.contain('login?logout');
```
Comment (rquintana-abstracta) [inline tests/logout.test.js]: Consider using Webdriver.io's **matchers** whenever possible for implementing asserts that verify elements or browsers properties, as they are more idiomatic and typically shorter / more readable expressions. For this case, something like:

```javascript
expect(await browser).toHaveUrlContaining('login?logout');
```
Reference: https://webdriver.io/docs/api/expect-webdriverio/#tohaveurlcontaining

File: tests/logout.test.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      await loginPage.open('login');
+      let username = 'jsmith@demo.io';
+      let password = 'Demo123!';
+      await loginPage.logIn(username, password);
+
+      await logoutPage.clickAvatar();
+      
+      await browser.back();
+      
+      expect(await browser.getUrl()).to.contain('login');
```
Comment (rquintana-abstracta) [inline tests/logout.test.js]: Same comment about using wdio's matchers for this

**Review by rquintana-abstracta — CHANGES_REQUESTED** [review]
Implement requested changes

File: pages/logout.page.js (hunk 1 above)
Comment (carperimb) [reply pages/logout.page.js]: image
that's the full name of the class...

File: tests/login.test.js (hunk 2 above)
Comment (carperimb) [reply tests/login.test.js]: Login TCs were skipped just for this sprint, as I mentioned in the description of this pull request...
Is there another way to run the logout tests only?

File: tests/logout.test.js (hunk 4 above)
Comment (carperimb) [reply tests/logout.test.js]: Error:
Invalid Chai property: toHaveUrlContaining

File: tests/logout.test.js (hunk 5 above)
Comment (carperimb) [reply tests/logout.test.js]: Same error:
Invalid Chai property: toHaveUrlContaining

File: tests/logout.test.js (hunk 4 above)
Comment (carperimb) [reply tests/logout.test.js]: image

Comment (carperimb) [reply tests/logout.test.js]: image

Comment (carperimb) [reply tests/logout.test.js]: What should I use: Chai or WebdriverIO??

File: tests/login.test.js (hunk 2 above)
Comment (carperimb) [reply tests/login.test.js]: I think the login tests were shown last sprint so there's no need to show them again...

File: tests/logout.test.js (hunk 3 above)
Comment (carperimb) [reply tests/logout.test.js]: Done

File: pages/logout.page.js (hunk 1 above)
Comment (rquintana-abstracta) [reply pages/logout.page.js]: Those are 2 classes (multiple css classes adding in html elements is achieved by separating each one with an empty space, so you add classes to an element by adding them after an empty space). If only one html element has one class, you can still find it by targeting that specific class with a css selector (like in this case, **user-avatar**)

File: tests/login.test.js (hunk 2 above)
Comment (rquintana-abstracta) [reply tests/login.test.js]: The repository is tipically accumulative (you are building a general regression suite) and execution patterns are typically managed at a different level (for example, using a CLI command with a specific flag signaling what suite you'd like to execute). This usage of **skip** and **only** is more typically used for debugging purposes, but removed when the feature is finished.

File: tests/logout.test.js (hunk 4 above)
Comment (rquintana-abstracta) [reply tests/logout.test.js]: According to the doc you've shared, expect-webdriverio is a super-set of chai's expect, therefore by using expect-webdriverio istead of chai's you'd achieve the best of both worlds. Go with globally overriding **expect** with expect-webdriverio implementation

File: tests/logout.test.js (hunk 5 above)
Comment (rquintana-abstracta) [reply tests/logout.test.js]: Same comment as above, globally override **expect** by using expect-webdriverio implementation

File: pages/logout.page.js (hunk 1 above)
Comment (carperimb) [reply pages/logout.page.js]: Error:
element ("[class="user-avatar"]") still not clickable after 10000ms

File: tests/logout.test.js (hunk 4 above)
Comment (carperimb) [reply tests/logout.test.js]: Done

File: tests/logout.test.js (hunk 5 above)
Comment (carperimb) [reply tests/logout.test.js]: Done

File: pages/logout.page.js (hunk 1 above)
Comment (rquintana-abstracta) [reply pages/logout.page.js]: what if you try a raw css selector (like `.user-avatar`) like:
```javascript
$(".user-avatar")
```

Comment (carperimb) [reply pages/logout.page.js]: Done:
get userAvatar(){return $('.user-avatar');}

File: tests/logout.test.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  beforeEach(() => {
+      loginPage.open('login');
+      let username = 'jsmith@demo.io';
+      let password = 'Demo123!';
+      loginPage.logIn(username, password);
+
+  })
+
+    it('Should show logout success message when clicking on logout menu', async () => {
+
```
Comment (rquintana-abstracta) [inline tests/logout.test.js]: There's too much space between lines here. Put them closer together.

File: tests/logout.test.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+      await logoutPage.clickAvatar();
+      
+      
+      expect(await browser).toHaveUrlContaining('login?logout');
+      
+
+    });
+    it('Should show login page when clicking browser back button after logging out the session', async () => {
+      
+
```
Comment (rquintana-abstracta) [inline tests/logout.test.js]: Same here, put lines closer together

File: tests/login.test.js (hunk 2 above)
Comment (carperimb) [reply tests/login.test.js]: Done!

File: wdio.conf.js [hunk 8]
```diff
@@ -74,13 +74,13 @@ exports.config = {
     // ===================
     // Define all options that are relevant for the WebdriverIO instance here
     //
-    beforeTest: function (test, context) {
+    /* beforeTest: function (test, context) {
```
Comment (rquintana-abstracta) [inline wdio.conf.js]: Delete these lines, don't comment them.

File: tests/logout.test.js (hunk 7 above)
Comment (carperimb) [reply tests/logout.test.js]: done

File: tests/logout.test.js (hunk 6 above)
Comment (carperimb) [reply tests/logout.test.js]: done

File: wdio.conf.js (hunk 8 above)
Comment (carperimb) [reply wdio.conf.js]: done

**Review by rquintana-abstracta — APPROVED** [review]

**Review by rquintana-abstracta — APPROVED** [review]

**Pull request merged**